sphinx_gallery/scrapers.py

```diff
@@ -230,11 +238,37 @@ def _anim_rst(anim, image_path, gallery_conf):
     else:
         writer = None
     anim.save(str(image_path), writer=writer, dpi=use_dpi)
-    html = anim._repr_html_()
-    if html is None:  # plt.rcParams['animation.html'] == 'none'
+
+    _, fmt = gallery_conf["matplotlib_animations"]
+    html = None
+    if fmt is None:
+        html = anim._repr_html_()
+        if html is None:  # plt.rcParams['animation.html'] == 'none'
+            html = anim.to_jshtml()
+    elif fmt == "html5":
+        html = anim.to_html5_video()
+    elif fmt == "jshtml":
         html = anim.to_jshtml()
-    html = indent(html, "     ")
-    return _ANIMATION_RST.format(html)
+    if html is not None:
+        html = indent(html, "     ")
+        return _ANIMATION_RST.format(html)
+
+    video = image_path.with_suffix(f".{fmt}")
+    anim.save(video)
+    options = ["autoplay"]
+    if getattr(anim, "_repeat", False):
+        options.append("loop")
+    dpi = rcParams["savefig.dpi"]
+    if dpi == "figure":
+        dpi = fig.dpi
+    video_uri = video.relative_to(gallery_conf["src_dir"]).as_posix()
+    html = _ANIMATION_VIDEO_RST.format(
```

Comment:
Question, in what cases would we use `_ANIMATION_VIDEO_RST` instead of `_ANIMATION_RST`, i.e., when is `html` `None` ?

Reply:
`html` is `None` if `fmt not in (None, "html5", "jshtml")`; those are the originally supported formats kept for backwards compatibility.